pyswarms/single/global_best.py

```diff
@@ -153,10 +157,14 @@ def optimize(self, objective_func, iters, print_step=1, verbose=1):
         tuple
             the global best cost and the global best position.
         """
+
+        cli_print("Arguments Passed to Objective Function: \nargs: %s \nkwargs: %s\n" % (args, kwargs),
+                  verbose, 2, logger=self.logger)
+
         for i in range(iters):
             # Compute cost for current position and personal best
-            self.swarm.current_cost = objective_func(self.swarm.position)
-            self.swarm.pbest_cost = objective_func(self.swarm.pbest_pos)
+            self.swarm.current_cost = objective_func(self.swarm.position, *args, **kwargs)
+            self.swarm.pbest_cost = objective_func(self.swarm.pbest_pos, *args, **kwargs)
             self.swarm.pbest_pos, self.swarm.pbest_cost = compute_pbest(
```

Comment:
In my opinion, this is just the modification we need. That is, to provide flexibility in the objective_function by providing an optional `*args` or `**kwargs` argument. There's no need to put `args = ()` or `kwargs = {}` in the main `optimize()` method.

Reply:
Sure! I fixed the optimize prototypes. The only goofy thing there is the *args had to be listed before the other key word arguments i.e. ` def fun(foo, bar, *args, baz=3, **kwargs)`